sktime/pipeline/pipeline.py

```diff
@@ -0,0 +1,584 @@
+"""class that implements a graph pipeline."""
+import weakref
+from copy import copy, deepcopy
+
+from sktime.base import BaseEstimator
+from sktime.pipeline.step import Step
+from sktime.transformations.series.subset import ColumnSelect
+
+__author__ = ["benHeid"]
+__all__ = ["Pipeline"]
+
+
+class MethodNotImplementedError(Exception):
+    """MethodNotImplementedError.
+
+    Exception class to raise if the required method is not supported
+    by the current pipeline instance since there is at least one skobject
+    in the pipeline that do not support transform and the required method.
+    """
+
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class Pipeline(BaseEstimator):
+    """Implementation of a Graphpipeline.
+
+    This class is a generalized graph pipeline. Generalized means that it can
+    contain forecasters, classifiers, etc. The graph pipeline mean that the structure
+    is not linear. I.e., the each element of the pipeline can be the input of multiple
+    other steps and not only one sucessors.
+
+    **Acknowledgements**
+    This graphical pipeline is inspired by pyWATTS that is developed by the Institute
+    for Automation and Applied Informatics (IAI) at Karlsruhe Institute of Technology.
+    The implementation is supported by IAI and the author benHeid is funded by
+    HelmholtzAI.
+
+    `fit(y, X, *args)` - changes state by running `fit` on all sktime estimators and
```

Comment:
should this include polymorphism, or are we primarily focused on forecasters for now?

Reply:
The pipeline is also working for classifiers. Compare `test_classifier_regression` in test_pipeline_regression

Reply:
ah, great!